From ad9c6b9d3236ad96a1d0a79b5cdbd36f37128298 Mon Sep 17 00:00:00 2001
From: Andy Ross <andrew.j.ross@intel.com>
Date: Mon, 19 May 2014 16:22:21 -0700
Subject: [PATCH 037/117] gmin/atomisp: Fix default_camera_caps

This was thought to be a legacy API, because it was a hard-coded,
one-entry stub upstream.  But the atomisp driver still follows the
returned pointer and crashes if we don't duplicate the upstream
behavior exactly.

Signed-off-by: Andy Ross <andrew.j.ross@intel.com>
---
 .../x86/platform/intel-mid/atomisp_gmin_platform.c | 22 +++++++++++-----------
 1 file changed, 11 insertions(+), 11 deletions(-)

diff --git a/arch/x86/platform/intel-mid/atomisp_gmin_platform.c b/arch/x86/platform/intel-mid/atomisp_gmin_platform.c
index a79b486..724beb7 100644
--- a/arch/x86/platform/intel-mid/atomisp_gmin_platform.c
+++ b/arch/x86/platform/intel-mid/atomisp_gmin_platform.c
@@ -26,20 +26,20 @@ static const struct atomisp_platform_data pdata = {
 };
 
 /*
- *   struct atomisp_sensor_caps {
- *       int stream_num;
- *   };
- *   struct atomisp_camera_caps {
- *       int sensor_num;
- *       struct atomisp_sensor_caps sensor[MAX_SENSORS_PER_PORT];
- *   };
+ * Legacy/stub behavior copied from upstream platform_camera.c.  The
+ * atomisp driver relies on these values being non-NULL in a few
+ * places, even though they are hard-coded in all current
+ * implementations.
  */
 const struct atomisp_camera_caps *atomisp_get_default_camera_caps(void)
 {
-	/* This is near-legacy.  The camera_caps field is ultimately used
-	 * only in two spots in atomisp_cmd, one checks if it's ==1 and
-	 * the other if it's <2 (is 0 legal?). */
-	return NULL;
+	static const struct atomisp_camera_caps caps = {
+		.sensor_num = 1,
+		.sensor = {
+			{ .stream_num = 1, },
+		},
+	};
+	return &caps;
 }
 EXPORT_SYMBOL_GPL(atomisp_get_default_camera_caps);
 
-- 
1.9.3

